asf_core_data/notebooks/get_address_matching_sample.py

```diff
@@ -0,0 +1,318 @@
+# ---
+# jupyter:
+#   jupytext:
+#     cell_metadata_filter: -all
+#     comment_magics: true
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.15.2
+#   kernelspec:
+#     display_name: asf_core_data_kernel
+#     language: python
+#     name: asf_core_data_kernel
+# ---
+
+# %%
+import pandas as pd
+import os
+from datetime import datetime
+
+# %%
+from asf_core_data.pipeline.mcs.process import mcs_epc_joining as joining
+
+# %% [markdown]
+# ## Set local_core_data_dir where EPC data is stored
+
+# %%
+## Set your local ASF core data dir
+local_core_data_dir = ""
+
+## Output data will be saved here
+outputs_data_dir = os.path.abspath("../../outputs/data")
+
+# %% [markdown]
+# ## Read in data
+#
+# EPC data: 2023 Q2 complete, GB dedupl and preprocessed\
+# MCS data: mcs_installations_231009
+
+# %%
+epc_data = pd.read_csv(
```

Comment:
Yes I think you are correct because [line 139-144 in `pipeline/mcs/generate_mcs_data.py` ](https://github.com/nestauk/asf_core_data/blob/15ccd754dbc31c2a8ba4d84463cba7ebfe5ebf82/asf_core_data/pipeline/mcs/generate_mcs_data.py#L139) uses default `epcs` parameter, which is `None`, and sets `all_records=True`. Lines [293-294 in `pipleine/mcs/process/mcs_epc_joining.py`](https://github.com/nestauk/asf_core_data/blob/15ccd754dbc31c2a8ba4d84463cba7ebfe5ebf82/asf_core_data/pipeline/mcs/process/mcs_epc_joining.py#L293) show this would set the `epc_version` to `"preprocessed"`.